MUICore/muicore/mui_server.py

```diff
@@ -21,26 +19,38 @@
     Tracer,
     RecordSymbolicBranches,
 )
-import manticore.utils.log
-from manticore.core.worker import WorkerThread, DaemonThread
-
+from manticore.core.worker import WorkerThread, WorkerProcess
+from manticore.utils.log import CallbackStream, ManticoreContextFilter
 from manticore.utils.enums import StateStatus, StateLists
+from manticore.utils.helpers import deque
 
 import uuid
 import shutil
 from inspect import currentframe, getframeinfo
 
 from .native_utils import parse_native_arguments
 from .evm_utils import setup_detectors_flags
-from manticore.utils.log import CallbackStream
 
 
 class ManticoreWrapper:
     def __init__(self, mcore_object, mthread):
         self.uuid = uuid.uuid4().hex
         self.manticore_object = mcore_object
         self.thread = mthread
-        self.log_queue = collections.deque(maxlen=5000)
+        self.log_queue = (
+            mcore_object._manager.Queue(15000)
+            if mcore_object._worker_type == WorkerProcess
+            else deque(maxlen=5000)
+        )
+
+    def append_log(self, msg):
+        q = self.log_queue
+        try:
+            q.append(msg)
+        except AttributeError:
```

Comment:
reasoning for this code is in provided in the manticore repo: https://github.com/trailofbits/manticore/blob/master/manticore/core/worker.py#L306-L312

Reply:
I think this would also be good to have in the code itself, as a comment (also use the "permalink" that includes the commit for the line numbers so that if things change in the future, it will still be the same)